github/util.py

```diff
@@ -221,3 +222,82 @@ def retrieve_email(username: str):
         )
     )
 
+def _create_team(
+    github: GitHub,
+    organization_name: str,
+    team_name: str
+):
+    # passed GitHub object must have org. admin authorization to create a team
+    organization = github.organization(organization_name)
+    team_list = list(filter(lambda t: t.name == team_name, organization.teams()))
+    if team_list:
+        util.verbose("Team {name} already exist".format(name=team_name))
+        return
+
+    try:
+        organization.create_team(name=team_name)
+        util.info("Team {name} created".format(name=team_name))
+    except ForbiddenError as err:
+        util.fail("{msg} Cannot create team {name} in org {org} due to missing privileges".format(
+            msg=err,
+            name=team_name,
+            org=organization_name
+        ))
+
+def _add_user_to_team(
+    github: GitHub,
+    organization_name: str,
+    team_name: str,
+    user_name: str
+):
+    # passed GitHub object must have org. admin authorization to add a user to a team
+    organization = github.organization(organization_name)
+    team_list = list(filter(lambda t: t.name == team_name, organization.teams()))
```

Comment:
this is redundant coding.

please do the following:

- create a function 'retrieve_team_by_name'
  - if the team exists, the github team object is returned
  - if it does not, it could (to your preference) return None or raise an Exception (I'd prefer the first)
  - as team names should be unique, a single team instance is returned (not a sequence)
- use this function to replace the redundant team-lookups here and in the function above

Reply:
Done